Sources/BrowserServicesKit/PrivacyConfig/AppPrivacyConfiguration.swift

```diff
@@ -88,14 +96,71 @@ public struct AppPrivacyConfiguration: PrivacyConfiguration {
         default: return false
         }
     }
+    
+    private func rolloutEnabled(subfeature: any PrivacySubfeature,
+                                rollouts: [PrivacyConfigurationData.PrivacyFeature.Feature.Rollout],
+                                randomizer: (Range<Double>) -> Double) -> Bool {
+        let defsPrefix = "config.\(subfeature.parent.rawValue).\(subfeature.rawValue)"
+        if userDefaults.bool(forKey: "\(defsPrefix).\(Constants.enabledKey)") {
+            return true
+        }
+        
+        var willEnable = false
+        if let rolloutSize = userDefaults.value(forKey: "\(defsPrefix).\(Constants.lastRolloutCountKey)") as? Int {
+            guard rolloutSize != rollouts.count else { return false }
+            // Sanity check as we need at least two values to compute the new probability
+            guard rollouts.count > 1 else { return false }
+            
+            // If the user has seen the rollout before, and the rollout count has changed
+            // Try again with the new probability
+            let y = rollouts[rollouts.count - 1].percent
+            let x = rollouts[rollouts.count - 2].percent
+            let prob = (y - x) / (100.0 - y)
+            if randomizer(0..<1) < prob {
+                // enable the feature
+                willEnable = true
+            }
+        } else if rollouts.count == 1 {
+            // First time user sees feature, and only one rollout
+            willEnable = randomizer(0..<100) < rollouts.first!.percent
+        } else {
+            // First time user sees feature, and multiple rollouts
+            for i in 1..<rollouts.count {
+                let y = rollouts[i].percent
+                let x = rollouts[i - 1].percent
+                let prob = (y - x) / (100.0 - y)
+                if randomizer(0..<1) < prob {
+                    willEnable = true
+                    break
+                }
+            }
```

Comment:
I think we can just take the last step here and roll using that, right? Given steps `[5, 10, 15]`, we should just be able to take `15` and treat it as our threshold.

Reply:
The spec says to compute it in this manner: https://app.asana.com/0/1203137811378537/1204934761241565/f

Reply:
That's true, though it's outdated - the Android implementation is doing it the simpler way, and Bartek has a comment [here](https://app.asana.com/0/0/1204934761241565/1204955479205817/f) suggesting this simpler approach also.

Reply:
Gotcha. Missed that. Updated now.